RecoTracker/LSTCore/src/alpaka/MiniDoublet.h

```diff
@@ -11,185 +13,60 @@
-          moduleIndices_buf(allocBufWrapper<uint16_t>(devAccIn, nMemoryLoc, queue)),
-          nMDs_buf(allocBufWrapper<unsigned int>(devAccIn, nLowerModules + 1, queue)),
-          totOccupancyMDs_buf(allocBufWrapper<unsigned int>(devAccIn, nLowerModules + 1, queue)),
-          dphichanges_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          dzs_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          dphis_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          shiftedXs_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          shiftedYs_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          shiftedZs_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          noShiftedDphis_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          noShiftedDphiChanges_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          anchorX_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          anchorY_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          anchorZ_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          anchorRt_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          anchorPhi_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          anchorEta_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          anchorHighEdgeX_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          anchorHighEdgeY_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          anchorLowEdgeX_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          anchorLowEdgeY_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          anchorLowEdgePhi_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          anchorHighEdgePhi_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          outerX_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          outerY_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          outerZ_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          outerRt_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          outerPhi_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          outerEta_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          outerHighEdgeX_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          outerHighEdgeY_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          outerLowEdgeX_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)),
-          outerLowEdgeY_buf(allocBufWrapper<float>(devAccIn, nMemoryLoc, queue)) {
-      alpaka::memset(queue, nMDs_buf, 0u);
-      alpaka::memset(queue, totOccupancyMDs_buf, 0u);
-    }
 
-    inline MiniDoublets const* data() const { return &data_; }
-    inline void setData(MiniDoubletsBuffer& buf) { data_.setData(buf); }
-  };
+  GENERATE_SOA_LAYOUT(MiniDoubletsSoALayout,
```

Comment:
Yeah, that sounds good. We can't place the new header in `src` since `src/alpaka` is an independent package, so I'll put it in `RecoTracker/LSTCore/interface/`, just like for `TrackCandidatesSoA.h`